datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -694,6 +690,94 @@ impl PhysicalExpr for BinaryExpr {
         let mut s = state;
         self.hash(&mut s);
     }
+
+    /// For each operator, [`BinaryExpr`] has distinct ordering rules.
+    /// TODO: There may be rules specific to some data types (such as division and multiplication on unsigned integers)
+    fn get_ordering(&self, children: &[SortProperties]) -> SortProperties {
+        let (left_child, right_child) = (&children[0], &children[1]);
+        match self.op() {
+            Operator::Plus => left_child.add(right_child),
+            Operator::Minus => left_child.sub(right_child),
+            Operator::Gt | Operator::GtEq => left_child.gt_or_gteq(right_child),
+            Operator::Lt | Operator::LtEq => right_child.gt_or_gteq(left_child),
+            Operator::And => left_child.and(right_child),
+            _ => SortProperties::Unordered,
+        }
+    }
+}
+
+impl SortProperties {
```

Comment:
it is somewhat strange to put this impl into a different module than the definion of `SortProperties`

Maybe we could move all SortProperties related stuff to `datafusion/physical-expr/src/sort_properties.rs` or something